presto-session-property-managers/src/main/java/io/prestosql/plugin/session/AbstractSessionPropertyManager.java

```diff
@@ -26,15 +26,24 @@
 {
     protected abstract List<SessionMatchSpec> getSessionMatchSpecs();
 
-    @Override
-    public Map<String, String> getSystemSessionProperties(SessionConfigurationContext context)
+    private Map<String, String> getSessionProperties(SessionConfigurationContext context)
     {
         // later properties override earlier properties
         Map<String, String> combinedProperties = new HashMap<>();
         for (SessionMatchSpec sessionMatchSpec : getSessionMatchSpecs()) {
             combinedProperties.putAll(sessionMatchSpec.match(context));
         }
-        return ImmutableMap.copyOf(combinedProperties);
+        return combinedProperties;
+    }
+
+    @Override
+    public Map<String, String> getSystemSessionProperties(SessionConfigurationContext context)
+    {
+        return getSessionProperties(context)
+                .entrySet()
+                .stream()
+                .filter(entry -> !entry.getKey().contains("."))
```

Comment:
extract this condition as method: `!isCatalogSessionProperty()`

Reply:
Done